src/lib/components/Swap/Toolbar/index.tsx

```diff
@@ -25,7 +26,7 @@ export default function Toolbar({ disabled }: { disabled?: boolean }) {
   } = useSwapInfo()
   const isRouteLoading = state === TradeState.SYNCING || state === TradeState.LOADING
   const isAmountPopulated = useIsAmountPopulated()
-
+  const { type: wrapType, loading: wrapLoading } = useWrapCallback()
```

Comment:
I don't believe the state from the hook will be shared by multiple invocations of the hook. You'd need an atom to accomplish that.